src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -813,12 +823,12 @@ object GraphFrame extends Serializable with Logging {
           for (prev <- prev) assert(prev.columns.toSet.contains(name))
           (prev, prevNames)
         } else {
-          (Some(maybeJoin(prev, nestV(name))), prevNames :+ name)
+          (Some(maybeCrossJoin(prev, nestV(name))), prevNames :+ name)
         }
 
       case NamedEdge(name, AnonymousVertex, AnonymousVertex) =>
         val eRen = nestE(name)
-        (Some(maybeJoin(prev, eRen)), prevNames :+ name)
+        (Some(maybeCrossJoin(prev, eRen)), prevNames :+ name)
```

Comment:
We don't.  I think it might work to have duplicates.  But I'm on board with explicitly outlawing that since there's no reason user should have duplicate edge names in a query.  Will update...